cloud-sql/mysql/Program.cs

```diff
@@ -14,21 +14,56 @@
  * the License.
  */
 
+using CloudSql.Settings;
+using Google.Cloud.Diagnostics.AspNetCore;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.IO;
 
 namespace CloudSql
 {
     public class Program
     {
+        
+        public static AppSettings AppSettings { get; private set; }
+
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            BuildWebHost(args).Build().Run();
+        }
+        
+        public static IWebHostBuilder BuildWebHost(string[] args)
+        {
+            ReadAppSettings();
+
+            string port = Environment.GetEnvironmentVariable("PORT") ?? "8080";
+            string hostingUrl = $"http://0.0.0.0:{port}";
+            string localhostUrl = "http://localhost:5567";

```

Comment:
Should this be`$"http://localhost:port"`?

Reply:
Reusing the same port causes a "System.Net.Sockets.SocketException: Address already in use" error in Cloud Run and App Engine. By using "localhost:5567" the same URL can be used for both running locally and for Run-KestralTest.

Reply:
Ok.  Please add a comment to that effect.

Reply:
Oh, I think I've seen this problem before.  The issue with hard-coding 5567 is it doesn't respect what the user sets on the command line or via environment variables or in config files.

Program.cs in the getting-started samples demonstrates a solution that still respects all the other ways to set the port number.
https://github.com/GoogleCloudPlatform/getting-started-dotnet/blob/master/HelloWorld/Program.cs